server/pbench/bin/pbench-unpack-tarballs

```diff
@@ -107,79 +119,91 @@ while read size result ;do
     ntotal=$ntotal+1
 
     link=$(readlink -e $result)
-    if [ ! -f "$link" ] ;then
-        echo "$TS: $link does not exist" | tee -a $mail_content >&4
+    if [ -z "$link" ] ;then
+        echo "$TS: symlink target for $result does not exist" | tee -a $mail_content >&4
         nerrs=$nerrs+1
+        # FIXME: quarantine $result
         continue
     fi
     resultname=$(basename $result)
     resultname=${resultname%.tar.xz}
-    hostname=$(basename $(dirname $link))
 
-    # echo "Link=$link"
-    # echo "Resultname=$resultname"
-    # echo "Hostname=$hostname"
-    # continue
+    # XXXX - for now, if it's a duplicate name, just punt and avoid
+    # producing the error
+    if [ ${resultname%%.*} == "DUPLICATE__NAME" ] ;then
+        ndups=$ndups+1
+        # FIXME: quarantine $result
+        continue
+    fi
+
+    basedir=$(dirname $link)
+    hostname=$(basename $basedir)
 
     # make sure that all the relevant state directories exist
     mk_dirs $hostname
     # ... and a couple of other necessities.
     if [[ $? -ne 0 ]] ;then
-        echo "$TS: Creation of $hostname processing directories failed: code $status" | tee -a $mail_content >&4
+        echo "$TS: Creation of $hostname processing directories failed for $result: code $status" | tee -a $mail_content >&4
         nerrs=$nerrs+1
+        # FIXME: quarantine $result
         continue
     fi
     mkdir -p $INCOMING/$hostname
     if [[ $? -ne 0 ]] ;then
-        echo "$TS: Creation of $INCOMING/$hostname failed: code $status" | tee -a $mail_content >&4
+        echo "$TS: Creation of $INCOMING/$hostname failed for $result: code $status" | tee -a $mail_content >&4
+        nerrs=$nerrs+1
+        # FIXME: quarantine $result
+        continue
+    fi
+    incoming=$INCOMING/$hostname/$resultname
+    if [ -e $incoming ]; then
+        echo "$TS: Incoming result, $INCOMING/$hostname/$resultname/, already exists, skipping $result" | tee -a $mail_content >&4
         nerrs=$nerrs+1
+        # FIXME: quarantine $result
         continue
     fi
 
     mkdir -p $UNPACK_PATH/$hostname
     status=$?
     if [[ $status -ne 0 ]] ;then
-        echo "$TS: mkdir -p $UNPACK_PATH/$hostname failed: code $status" | tee -a $mail_content >&4
+        echo "$TS: mkdir -p $UNPACK_PATH/$hostname failed $result: code $status" | tee -a $mail_content >&4
         nerrs=$nerrs+1
+        # FIXME: quarantine $result
         continue
     fi
 
-    # XXXX - for now, if it's a duplicate name, just punt and avoid producing the error - the full
-    # solution will involve renaming the unpacked directory appropriately.
-    if [ ${resultname%%.*} == "DUPLICATE__NAME" ] ;then
-        ndups=$ndups+1
-        continue
-    fi
-
-    let start_time=$(date +%s)
+    let start_time=$(timestamp-seconds-since-epoch)
     pushd $UNPACK_PATH/$hostname > /dev/null 2>&1
     tar --extract --no-same-owner --no-overwrite-dir --touch --file="$result"
```

Comment:
Unpack into tmp and check that names match.

Reply:
Thanks, working on that now ...